mkt/games/paginator.py

```diff
@@ -0,0 +1,45 @@
+from django.core.paginator import EmptyPage, Page
+
+from mkt.api.paginator import ESPaginator
+
+
+class ESGameAggregationPaginator(ESPaginator):
+    """
+    Paginator that handles aggregated results using a hardcoded aggregation
+    name and bucket name specified in mkt.games.filters.
+
+    Normally, the ESPaginator looks for results in execute().hits, this will
+    look for results in execute().aggregations which follows a different
+    format.
+
+    Ideally, this class would be able to be configured with different
+    aggregation and bucket names, but there's not an easy way to pass in
+    context like we can for serializers.
+    """
+    def page(self, number):
+        """
+        Returns a page object.
+        """
+        result = self.object_list.execute()
+
+        # Pull the results from the aggregations.
+        hits = []
+        aggs = result.aggregations
+        buckets = aggs['top_hits']['buckets']
+        for bucket in buckets:
+            hits.append(bucket['first_game']['hits']['hits'][0])
+
+        page = Page(hits, number, self)
+
+        # Update the `_count`.
+        self._count = len(page.object_list)
```

Comment:
It might be better to extract the data in the view (using a custom `paginate_queryset` method or something similar) then. IMHO keeping a paginator that does not really paginate around is confusing.